hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -125,21 +122,16 @@ public ManagementCenterService(HazelcastInstanceImpl instance) {
         this.timedMemberStateFactory = new TimedMemberStateFactory(instance);
         this.identifier = newManagementCenterIdentifier();
 
-        registerListeners();
+        if (this.managementCenterConfig.isEnabled()) {
+            this.instance.getCluster().addMembershipListener(new ManagementCenterService.MemberListenerImpl());
+            start();
```

Comment:
So if I understand it correctly:

old: start mancenterservice once the HZ instance is started.

new: start mancenterservice before/during HZ instance started.

Can't this cause problems? If the mancenter is being exposed before the HZ instance is fully started; can't this lead to running into partially initialised data-structures?